app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Users
+  class OmniauthCallbacksController < ::Devise::OmniauthCallbacksController
+    def google_oauth2
+      @user = OmniauthSignInService.new.find_or_create_user(access_token: request.env['omniauth.auth'])
```

Comment:
just out of curiosity.. how does security works here? Seems like the email is going to be in plain text in the request, right? Why can't it be hacked?

Reply:
That MITM attack is avoided by whitelisting. When we create credentials on the google console we need to specify correct callback URLs, so we know that Google will only send requests to us and since we started the request we know we sent to Google, so no one can get between that. Plus, it's server-to-server communication, so it's very difficult to interfere.

Reply:
ahh gotcha, makes sense